crates/csharp/src/interface.rs

```diff
@@ -496,10 +586,40 @@ impl InterfaceGenerator<'_> {
             );
         }
 
+        if async_ {
+            let import_module_name = &self.resolve.name_world_key(interface_name.unwrap());
+
+            uwriteln!(
+                self.csharp_interop_src,
+                r#"
+            [global::System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute(EntryPoint = "[callback][async-lift]{import_module_name}#{wasm_func_name}")]
+            public static uint {camel_name}Callback(uint eventRaw, uint waitable, uint code)
+            {{
+                // TODO: decode the parameters
+                return (uint)CallbackCode.Exit;
+            }}
+            "#
+            );
+
+            let task_return_param = match &sig.results[..] {
```

Comment:
The task return function could accept up to 16 core parameters (e.g. 2 for a `string`, 5 for a `tuple<string, list<u8>, u32>`, etc.).

Reply:
I'm hoping I'll be forced to address all these shortcuts as I go through enabling the tests.  If not, then I've added a TODO here and I can address the todo's at the end.